### ISSUE TITLE
Cursor control in complete suggestion?

### ISSUE THREAD

**Issue by emicklei** [peterh/liner#128]
Hi,

I am looking for a way to move the cursor into a completed suggestion.
For example, if I type `p` and it completes to `play("")`, I would like the cursor to be positioned right after the first `"`.  

I tried adding an ANSI escape sequence `\033[2D` (move cursor back 2x) but it is not being interpreted. An alternative could be to have a post complete func and a way to tell the *State that a cursor move is requested (programmatic CtrlB).

Before working on such a feature, I would like to have some feedback on this.

Thx,
Ernest

**Comment by peterh** [comment]
The last time we needed more features, we upgraded `Completer` with `WordCompleter`. This time, it probably makes sense to upgrade `WordCompleter` to `PosCompleter`

Something like
```go
type WordAndPos struct {
    Word string // completion string
    Pos int // position of cursor within completion string
}
type PosCompleter func(line string, pos int) (head string, completions []WordAndPos, tail string)
```

(except with a better name than WordAndPos)

Thoughts?

**Comment by emicklei** [comment]
yes, I like this proposal. Finding a good name is always a challenge. 
Maybe to generalise this, instead of returning a Pos int, we could have a CursorInstruction that be used to do:
- set position (like WordPos)
- move word back
- set to end
- create a selection (future feature?)

**Comment by peterh** [comment]
"Move word back" and "set to end" are redundant when you have "set position". I especially want to avoid things like "move word back" when "word" is so poorly defined (and subject to change; See #113 for discussion)

liner isn't a text editor. I try to keep it as close to bash as possible. Unless bash has selections that I'm not aware of (which is possible, bash has lots of features that I'm not aware of) "create a selection" sounds like a feature that liner won't ever have.

**Comment by emicklei** [comment]
fair enough. let's keep it simple. Suggestion:

- type EditableWordCompleter
- func(line string, pos int) (head string, completions []EditableWord, tail string)

Rationale: putting the cursor "in" the completed word is for the purpose of editing it after completion

**Comment by peterh** [comment]
I don't know if I like that more or less than PosCompleter. As they say, there are only two hard things in programming: Naming, Caching, and off-by-one errors.

You're doing the work, I'll let you choose the name (within reason).

**Comment by emicklei** [comment]
work in progress https://github.com/emicklei/liner/tree/poscomplete
